RFS/src/main/java/com/rfs/common/http/SigV4AuthTransformer.java

```diff
@@ -0,0 +1,52 @@
+package com.rfs.common.http;
+
+import java.time.Clock;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.opensearch.migrations.IHttpMessage;
+import org.opensearch.migrations.aws.SigV4Signer;
+
+import reactor.core.publisher.Mono;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+
+public class SigV4AuthTransformer implements AuthTransformer {
+    private final SigV4Signer signer;
+
+    public SigV4AuthTransformer(AwsCredentialsProvider credentialsProvider, String service, String region, String protocol, Supplier<Clock> timestampSupplier) {
+        this.signer = new SigV4Signer(credentialsProvider, service, region, protocol, timestampSupplier);
+    }
+
+    @Override
+    public Mono<TransformedRequest> transform(String method, String path, Map<String, List<String>> headers, Mono<String> body) {
```

Comment:
Is there a reason that body is a string? I think we should be able to sign any body type such as binary / compressed data.

Reply:
Body was a string for existing rest client compatibility. I've since moved it to a bytebuffer